File: 242-validAnagram.js
function isAnagram(s, t) {
    if (s.length !== t.length) {
        return false;
    } 
    let countObj = {};
    for (let i = 0; i <= s.length; i++) {
        if (s[i] in countObj) {
            countObj[s[i]]++;
        } else {
            countObj[s[i]] = 1;
        }
    }
    for (let i = 0; i <= t.length; i++) {
        if (t[i] in countObj) {
            countObj[t[i]]--;
        } else {
            countObj[t[i]] = 1;
        }
    }
    for (let key in countObj) {
        if (countObj[key] !== 0) {
            return false;
        }
    }
    return true;

}

File: 242-validAnagram.py
def isAnagram(s,t): 
    if len(s) != len(t):
        return False
    count_char = {}
    for i in range(len(s)): 
        if (s[i] in count_char):
            count_char[s[i]] += 1
        else: 
            count_char[s[i]] = 1
    for i in range(len(t)):
        if (t[i] in count_char):
            count_char[t[i]] -= 1
    
    for key in count_char: 
        if (count_char[key] != 0):
            return False
    
    return True

print(isAnagram("rat", "car"))
